internal/olm/operator/registry/operator_installer_test.go

```diff
@@ -16,22 +16,233 @@ package registry
 
 import (
 	"context"
+	"fmt"
+	// TODO(asmacdo) why doesnt vimgo do this for me
+	"os"
 
+	// "github.com/fatih/color"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	v1 "github.com/operator-framework/api/pkg/operators/v1"
 	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	// apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	// "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	"github.com/operator-framework/operator-sdk/internal/olm/operator"
+	"github.com/operator-framework/operator-sdk/internal/util/k8sutil"
 )
 
+type FakeCatalogCreator struct {
+	client crclient.Client
+}
+
+func (f FakeCatalogCreator) CreateCatalog(ctx context.Context, name string) (*v1alpha1.CatalogSource, error) {
+	cs := &v1alpha1.CatalogSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "MockCatalogSource",
+		},
+	}
+	if err := f.client.Create(ctx, cs); err != nil {
+		return nil, fmt.Errorf("Unable to create CatalogSource with fake client: %v", err)
+	}
+	return cs, nil
+}
+
 var _ = Describe("OperatorInstaller", func() {
-	Describe("InstallOperator", func() {
-		// TODO: fill this in once run bundle is done
+	Describe("NewOperatorInstaller", func() {
+		It("should create an OperatorInstaller", func() {
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+			// TODO(asmacdo) Is there anything I can test here that wouldnt be caught by the compiler?
+			// TODO(asmacdo)It doesnt even throw an err.
+			_ = NewOperatorInstaller(cfg)
+		})
+	})
+
+	Describe("createSubscription", func() {
+		var (
+			oi  *OperatorInstaller
+			sch *runtime.Scheme
+		)
+		BeforeEach(func() {
+			// Setup and fake client
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
```

Comment:
this is weird. We shouldn't have to use a real kubeconfig since we're using a fake client. Should be able to do the following:

```golang
cfg := &operator.Configuration{}
...
cfg.Client = fake.NewFakeClientWithScheme(sch)
```